x-pack/plugins/ml/public/application/components/navigation_menu/main_tabs.tsx

```diff
@@ -66,59 +69,116 @@ function getTabs(disableLinks: boolean): Tab[] {
 }
 interface TabData {
   testSubject: string;
-  pathId?: string;
+  pathId?: MlUrlGeneratorState['page'];
+  name: string;
 }
 
 const TAB_DATA: Record<TabId, TabData> = {
-  overview: { testSubject: 'mlMainTab overview' },
+  overview: {
+    testSubject: 'mlMainTab overview',
+    name: i18n.translate('xpack.ml.overviewTabLabel', {
+      defaultMessage: 'Overview',
+    }),
+  },
   // Note that anomaly detection jobs list is mapped to ml#/jobs.
-  anomaly_detection: { testSubject: 'mlMainTab anomalyDetection', pathId: 'jobs' },
-  data_frame_analytics: { testSubject: 'mlMainTab dataFrameAnalytics' },
-  datavisualizer: { testSubject: 'mlMainTab dataVisualizer' },
-  settings: { testSubject: 'mlMainTab settings' },
-  'access-denied': { testSubject: 'mlMainTab overview' },
+  anomaly_detection: {
+    testSubject: 'mlMainTab anomalyDetection',
+    name: i18n.translate('xpack.ml.anomalyDetectionTabLabel', {
+      defaultMessage: 'Anomaly Detection',
+    }),
+    pathId: 'jobs',
+  },
+  data_frame_analytics: {
+    testSubject: 'mlMainTab dataFrameAnalytics',
+    name: i18n.translate('xpack.ml.dataFrameAnalyticsTabLabel', {
+      defaultMessage: 'Data Frame Analytics',
+    }),
+  },
+  datavisualizer: {
+    testSubject: 'mlMainTab dataVisualizer',
+    name: i18n.translate('xpack.ml.dataVisualizerTabLabel', {
+      defaultMessage: 'Data Visualizer',
+    }),
+  },
+  settings: {
+    testSubject: 'mlMainTab settings',
+    name: i18n.translate('xpack.ml.settingsTabLabel', {
+      defaultMessage: 'Settings',
+    }),
+  },
+  'access-denied': {
+    testSubject: 'mlMainTab overview',
+    name: i18n.translate('xpack.ml.accessDeniedTabLabel', {
+      defaultMessage: 'Access Denied',
+    }),
+  },
 };
 
 export const MainTabs: FC<Props> = ({ tabId, disableLinks }) => {
+  const {
+    services: {
+      chrome: { docTitle },
+    },
+  } = useMlKibana();
   const [globalState] = useUrlState('_g');
   const [selectedTabId, setSelectedTabId] = useState(tabId);
   function onSelectedTabChanged(id: TabId) {
     setSelectedTabId(id);
   }
 
   const tabs = getTabs(disableLinks);
+  const mlUrlGenerator = useMlUrlGenerator();
+  const navigateToPath = useNavigateToPath();
+
+  const redirectToTab = async (defaultPathId: MlUrlGeneratorState['page']) => {
+    const pageState =
+      globalState?.refreshInterval !== undefined
+        ? {
+            globalState: {
+              refreshInterval: globalState.refreshInterval,
+            },
+          }
+        : undefined;
+    // @ts-ignore
+    const path = await mlUrlGenerator.createUrl({
+      page: defaultPathId,
+      // only retain the refreshInterval part of globalState
+      // appState will not be considered.
+      pageState,
+    });
+
+    await navigateToPath(path, false);
+  };
+
+  useEffect(() => {
+    docTitle.change([TAB_DATA[selectedTabId].name, ML_APP_NAME, 'Elastic']);
```

Comment:
I am seeing a duplicate 'Elastic' for all my tab titles:

![image](https://user-images.githubusercontent.com/7405507/92764503-63968000-f38c-11ea-91d7-ec922f8f07c3.png)

Reply:
Fixed here https://github.com/elastic/kibana/pull/76735/commits/dc3ba76792563da7c40db99ac11c6d69100f3bd7